### PR TITLE
[Lock] Add an auto release tips

### PR DESCRIPTION
**Pull request by jderusse**

### CODE REVIEW

File: components/lock.rst [hunk 1]
```diff
@@ -62,6 +62,13 @@ method can be safely called repeatedly, even if the lock is already acquired.
     to be used by several services, they should share the same ``Lock`` instance
     returned by the ``Factory::createLock`` method.
 
+.. tip::
+    While it's recommended to release the ``Lock`` as soon as it is no longer
```
Comment (xabbuh) [inline components/lock.rst]: I would reword the paragraph a bit:

```rst
.. tip::
    While it's recommended to release the lock as soon as it is no longer
    required, it will automatically be released on instance destruction. In
    some cases, it can be useful to lock a resource across several requests.
    To disable the automatic release behavior, set the third argument of
    the ``createLock()`` method to ``false``.
```

**Comment by javiereguiluz** [comment]
Thanks Jérémy.
